### ISSUE TITLE
Products Block: Enabled the `Inherit query from template` by default when adding the block to archive template

### ISSUE THREAD

**Issue by dinhtungdu** [woocommerce/woocommerce-gutenberg-products-block#8216]
## Is your feature request related to a problem? Please describe.
When adding the Products block to product archive templates, we should enable `Inherit query from template` by default. The Query Loop block does the same when adding to Archive templates.

## Additional context
cc @sunyatasattva as you lead the Products block, what do you think about this?

**Comment by sunyatasattva** [comment]
Totally agree. I see you have added it to the #8059 Epic, so I assume it's going to be on your backlog, right? Or would you rather us add it to the PQ Epic?

**Comment by dinhtungdu** [comment]
> I see you have added it to the https://github.com/woocommerce/woocommerce-blocks/issues/8059 Epic, so I assume it's going to be on your backlog

Yes, @albarin is looking at this issue already.

**Comment by albarin** [comment]
@dinhtungdu @sunyatasattva quick question: should the `Inherit query from template` option be set to true and disabled or users should have the option to set it to false? 🤔 Thanks!

**Comment by kmanijak** [comment]
> should the Inherit query from template option be set to true and disabled or users should have the option to set it to false? 

@albarin, I have discussed that with @dinhtungdu some time ago and here's the outcome of this discussion:

> The Products block representing the main loop must have the Inherit query from template setting enabled. But that setting can be toggled on/off by merchants which leads to unexpected behavior. We should inform merchants about that misconfiguration to prevent confusion.

1. User **can** toggle on/off the option. A use case for that is when a user wants to include more than one Products blocks: 
  - a main one - option to inherit from global query enabled by default
  - an additional one - for example for featured products. In this case the user must be able to disable the option to inherit from global query.
2. Comment mentions adding a notification that neither of Products block on the page inherits from global query, but I remember we discussed that as a nice-to-have feature that may be added in next iterations (although @dinhtungdu should confirm once he's back. Also, happy to hear @sunyatasattva opinion).
